symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/activity/GifActivityMain.java

```diff
@@ -0,0 +1,86 @@
+package com.symphony.bdk.examples.activity;
+
+import static com.symphony.bdk.core.config.BdkConfigLoader.loadFromSymphonyDir;
+
+import com.symphony.bdk.core.SymphonyBdk;
+import com.symphony.bdk.core.activity.AbstractActivity;
+import com.symphony.bdk.core.activity.ActivityRegistry;
+import com.symphony.bdk.core.activity.command.SlashCommand;
+import com.symphony.bdk.core.activity.model.ActivityInfo;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * <p>Activity API demonstration around a simple Gif bot logic.</p>
+ * <br/>
+ * <h3>
+ * Available commands
+ * </h3>
+ * <table>
+ *   <tr>
+ *     <td><pre>@BotMention /gif</pre></td>
+ *     <td>Displays the Gif category Elements form</td>
+ *   </tr>
+ *   <tr>
+ *     <td><pre>@BotMention /gif category</pre></td>
+ *     <td>Direct command including Gif category as parameter</td>
+ *   </tr>
+ * </table>
+ */
+@Slf4j
+public class GifActivityMain {
+
+  public static void main(String[] args) throws Exception {
+
+    // setup SymphonyBdk facade object
+    final SymphonyBdk bdk = new SymphonyBdk(loadFromSymphonyDir("config.yaml"));
+
+    // register new activity that sends a gif to the chat
+    bdk.activities().register(new GifCommand());
+
+    // displays the Gif form on /gif command with no params
+    bdk.activities().register(new SlashCommand("/gif", true, context ->
+        // send message contains Elements form to select a gif category
+        bdk.messages().send(context.getStreamId(), loadGifElementsForm())
+    ));
+
+    // register a "formReply" activity that handles the Gif category form submission
+    bdk.activities().register(new GifFormReplyActivity());
+
+    // display activities documentation in the console
+    reportActivities(bdk.activities());
+
+    // finally, start the datafeed loop
+    bdk.datafeed().start();
```

Comment:
why do I still have to bother with datafeed :P 
why not bdk.activities().start()? (that would delegate) but would make more sense as you would not know that activities and datafeed are related

Also why activities? is that a concept we already have? (vs commands() for instance)

Reply:
> why do I still have to bother with datafeed 

Yes I think that we could auto-start the datafeed from the `SymphonyBdk` constructor. We will talk about it later this Sprint with `APP-2908` 

> why not bdk.activities().start()? (that would delegate) but would make more sense as you would not know that activities and datafeed are related

I'll clearly document that datafeed and activities are related, that's not something we want to hide to end-developers

> Also why activities? is that a concept we already have? (vs commands() for instance)

Activities is a concept that gather user commands as well as form replies. We decided to name this new concept like this in order to sounds _BPMN-like_ feature (https://www.visual-paradigm.com/guide/bpmn/bpmn-activity-types-explained/), as this is also strongly related to workflow implementation.

Reply:
👍